extensions/tensorflow/TFExtractTopLabels.cpp

```diff
@@ -133,8 +136,11 @@ int64_t TFExtractTopLabels::LabelsReadCallback::process(const std::shared_ptr<io
 
   while (total_read < stream->size()) {
     auto read = stream->read(reinterpret_cast<uint8_t *>(&buf[0]), static_cast<int>(buf_size));
+    if (read < 0) {
```

Comment:
This will need to be changed after my PR #1028 is merged, but it might not conflict, because this is new code. If this slipped through, it would mean that `read` is never smaller than zero and line 143 becomes a loop to the largest `size_t` value.

Could you specify the type on line 138 to be int and use brace initialization (direct-list-initialization) to make this an error when `stream->read()` returns a `size_t`? Or add a static_assert that the variable is indeed of type int.

Reply:
Updated in 0185ef84c7e1855c3f73fecf608e37aa4bf66ef5 I'll update it after the merge